src/fn_call.rs

```diff
@@ -213,14 +215,24 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
                 // figure out some way to actually process some of them.
                 //
                 // `libc::syscall(NR_GETRANDOM, buf.as_mut_ptr(), buf.len(), GRND_NONBLOCK)`
-                // is called if a `HashMap` is created the regular way.
-                match this.read_scalar(args[0])?.to_usize(this)? {
-                    318 | 511 => {
-                        return err!(Unimplemented(
-                            "miri does not support random number generators".to_owned(),
-                        ))
+                // is called if a `HashMap` is created the regular way (e.g. HashMap<K, V>).
+                match (this.read_scalar(args[0])?.to_usize(this)? as i64, tcx.data_layout.pointer_size.bits()) {
+                    // SYS_getrandom on x86_64 and x86 respectively
+                    (318, 64) | (355, 32) => {
```

Comment:
Uh, what a hack.

Aren't these available as constants somewhere, so that we could do what @oli-obk did for `sysconf`?

Reply:
Yes, `SYS_getrandom` seems to be a thing in libc. Can you make it compare with that?

Reply:
Unfortunately, you can only get `SYS_getrandom` [for the current platform](https://docs.rs/libc/0.2.51/libc/constant.SYS_getrandom.html). We need to get it for both 32-bit and 64-it Linux, which I don't believe `libc` exposes.

Reply:
Fixed

Reply:
Yes that's what I mean. Awesome :)